src/test/java/spoon/test/serializable/ModelStreamerTest.java

```diff
@@ -0,0 +1,80 @@
+package spoon.test.serializable;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import spoon.Launcher;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.visitor.Filter;
+import spoon.support.CompressionType;
+import spoon.support.SerializationModelStreamer;
+
+public class ModelStreamerTest {
+
+	private final String filename = "./src/test/resources/serialization/factory.ser";
+
+	private Factory factory;
+
+	private File file;
+
+	@Before
+    public void setUp() {
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/main/java/spoon/reflect/declaration");
+		launcher.buildModel();
+		this.factory = launcher.getFactory();
+		file = new File(filename);
+		file.deleteOnExit();
+    }
+
+	@Test
+	public void testGZipCompression() throws IOException {
+
```

Comment:
I would suggest to specify explicitely that you're using GZIP compression type: the default value might evolve in the future.

Reply:
You're right! I'll do that tonight after work.